src/database-controller/src/watcher/framework/index.js

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+require('module-alias/register');
+require('dotenv').config();
+const fetch = require('node-fetch');
+const AsyncLock = require('async-lock');
+const logger = require('@dbc/core/logger');
+const { getFrameworkInformer } = require('@dbc/core/k8s');
+const { alwaysRetryDecorator } = require('@dbc/core/util');
+const config = require('@dbc/watcher/framework/config');
+
+const lock = new AsyncLock({ maxPending: Number.MAX_SAFE_INTEGER });
+
+async function synchronizeFramework(eventType, apiObject) {
+  const res = await fetch(
+    `${config.writeMergerUrl}/api/v1/watchEvents/${eventType}`,
+    {
+      method: 'POST',
+      body: JSON.stringify(apiObject),
+      headers: { 'Content-Type': 'application/json' },
+      timeout: config.writeMergerConnectionTimeoutSecond * 1000,
+    },
+  );
+  if (!res.ok) {
+    throw new Error(`Request returns a ${res.status} error.`);
+  }
+}
```

Comment:
In case k8s too many events(pods/frameworks/events), to be more efficient.

Can we move watcher inside WriteMerge process, instead of a standalone process needs RPC call?

Reply:
For example, other watcher (pods/events) also directly touch db. We can align with framework watcher

---
In reply to: [466199062](https://github.com/microsoft/pai/pull/4761#discussion_r466199062) [](ancestors = 466199062)

Reply:
As in my previous experiment, bottleneck is CPU utilization, not DB writing.

I suggest to start working on the optimization-related items after we have done the stress test.

Reply:
Fine to first do stress test.
But maybe it is the (De)Serialization between the 2 http process, caused the high CPU utilization.
So if you merge them in one process, no (De)Serialization overheads.

Reply:
Yes, it could be.